src/components/UserSelector.tsx

```diff
@@ -1,16 +1,64 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import { getAllUsers } from '../api/users';
+import { User } from '../types/User';
+
+type Props = {
+  onSelectUser: (value: User) => void;
+  selectedUser: User | null;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  onSelectUser,
+  selectedUser,
+}) => {
+  const [dropDownVisible, setDropDownVisible] = useState(false);
+  const [userList, setUserList] = useState<User[] | null>(null);
+  const coverRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    getAllUsers().then(res => {
+      setUserList(res);
+    });
+  }, []);
+
+  /* event.target.className !== 'button' || */
+
+  useEffect(() => {
+    const closeDropDown = (e: any) => {
+      if (
+        dropDownVisible &&
+        coverRef.current &&
+        !coverRef.current?.contains(e.target)
+      ) {
+        setDropDownVisible(false);
+      }
+    };
+
+    document.addEventListener('click', closeDropDown);
+
+    return () => removeEventListener('click', closeDropDown);
```

Comment:
I would create a hook `useOutsideClickListener` and use it here, just to keep this logic out of the dropdown itself

See for instance here: https://www.30secondsofcode.org/react/s/use-click-outside/

Reply:
Your variant is much shorter) 
I definitely will try it now)
Thank you!)

Reply:
Please have a look if I typed Mouse event correctly?

`const useClickOutside = (
  ref: React.RefObject<HTMLButtonElement>,
  callback: () => void,
) => {
  const handleClick = (e: MouseEvent) => {
    if (ref.current && !ref.current.contains(e.target as Node)) {
      callback();
    }
  };

  useEffect(() => {
    document.addEventListener('click', handleClick);

    return () => {
      document.removeEventListener('click', handleClick);
    };
  });
};`